phpBB/includes/acp/acp_board.php

```diff
@@ -510,6 +510,29 @@ function main($id, $mode)
 			}
 		}
 
+		if ($mode == 'avatar' && $cfg_array['allow_avatar_upload'])
+		{
+			// If avatar uploading is enabled but the path setting is empty,
+			// config variable validation is bypassed. Catch the case here
+			if (!$cfg_array['avatar_path'])
+			{
+				$error[] = $language->lang('AVATAR_NO_UPLOAD_PATH');
+			}
+			else
```

Comment:
Change this to `else if (!$submit)`. The filesystem check is only needed then and the check for `$submit` can be removed from the next `if()`